src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -614,8 +614,20 @@ public bool IsHardforkEnabled(Hardfork hardfork)
         {
             if (PersistingBlock is null)
                 return true;
+
             if (!ProtocolSettings.Hardforks.TryGetValue(hardfork, out uint height))
-                return true;
+                return false;
+
+            // Pre-computed previous hardforks
+            var allHardforks = Enum.GetValues(typeof(Hardfork)).Cast<Hardfork>().ToList();
+            foreach (var hf in allHardforks)
+            {
+                if (hf >= hardfork)
+                    break;
+                if (!ProtocolSettings.Hardforks.ContainsKey(hf))
+                    return false;
+            }
```

Comment:
However, I'm not sure that we need this change. Consider the case when some Basilisk is set in the config, but Aspidochelone isn't. Then the node's behaviour is to silently disable Basilisk, although the user would expect it to be enabled (accoarding to the node config).

I consider we should exactly follow the user's config and if the config says that Basilisk is enabled and Aspidochelone is disabled, then let's run exactly this node configuration.

Reply:
this has to be this way, otherwise, we have no idea when the previous hardfork should be enabled.

Reply:
hardfork from my understanding is that from A-->B-->C--D... one hardfork after another, which means B has to be before C and A has to be before A if we wish to enable C.

Reply:
I thought that you can enable them independently, but not sure what was the original intention. IMO, it's a nice ability to enable/disable them independently.

Reply:
> I thought that you can enable them independently, but not sure what was the original intention. IMO, it's a nice ability to enable/disable them independently.

It is highly possible that different hardforks have inheritance relationships.